{"ast":null,"code":"var _jsxFileName = \"C:\\\\Users\\\\a.sutic\\\\React\\\\clothing\\\\client\\\\src\\\\components\\\\error-boundary\\\\error-boundary.component.jsx\";\nimport React, { Component } from \"react\";\n\nclass ErrorBoundary extends Component {\n  constructor() {\n    super();\n    this.setState = {\n      hasError: false\n    };\n  } // if children through an error that error is passed into this function\n\n\n  static getDerivedStateFromError(error) {\n    // process error\n    return {\n      hasError: true\n    };\n  }\n\n  componentDidCatch(error, info) {\n    console.log(error);\n  }\n\n  render() {\n    if (this.state.hasError) {\n      return React.createElement(\"div\", {\n        __source: {\n          fileName: _jsxFileName,\n          lineNumber: 23\n        },\n        __self: this\n      }, \"Something went wrong\");\n    }\n\n    return this.props.children;\n  }\n\n}\n\nexport default ErrorBoundary;","map":{"version":3,"sources":["C:\\Users\\a.sutic\\React\\clothing\\client\\src\\components\\error-boundary\\error-boundary.component.jsx"],"names":["React","Component","ErrorBoundary","constructor","setState","hasError","getDerivedStateFromError","error","componentDidCatch","info","console","log","render","state","props","children"],"mappings":";AAAA,OAAOA,KAAP,IAAgBC,SAAhB,QAAiC,OAAjC;;AAEA,MAAMC,aAAN,SAA4BD,SAA5B,CAAsC;AACpCE,EAAAA,WAAW,GAAG;AACZ;AACA,SAAKC,QAAL,GAAgB;AACdC,MAAAA,QAAQ,EAAE;AADI,KAAhB;AAGD,GANmC,CAQpC;;;AACA,SAAOC,wBAAP,CAAgCC,KAAhC,EAAuC;AACrC;AACA,WAAO;AAAEF,MAAAA,QAAQ,EAAE;AAAZ,KAAP;AACD;;AAEDG,EAAAA,iBAAiB,CAACD,KAAD,EAAQE,IAAR,EAAc;AAC7BC,IAAAA,OAAO,CAACC,GAAR,CAAYJ,KAAZ;AACD;;AAEDK,EAAAA,MAAM,GAAG;AACP,QAAG,KAAKC,KAAL,CAAWR,QAAd,EAAwB;AACpB,aAAO;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,gCAAP;AACH;;AAED,WAAO,KAAKS,KAAL,CAAWC,QAAlB;AACD;;AAxBmC;;AA2BtC,eAAeb,aAAf","sourcesContent":["import React, { Component } from \"react\";\r\n\r\nclass ErrorBoundary extends Component {\r\n  constructor() {\r\n    super();\r\n    this.setState = {\r\n      hasError: false\r\n    };\r\n  }\r\n\r\n  // if children through an error that error is passed into this function\r\n  static getDerivedStateFromError(error) {\r\n    // process error\r\n    return { hasError: true };\r\n  }\r\n\r\n  componentDidCatch(error, info) {\r\n    console.log(error);\r\n  }\r\n\r\n  render() {\r\n    if(this.state.hasError) {\r\n        return <div>Something went wrong</div>\r\n    }\r\n\r\n    return this.props.children;\r\n  }\r\n}\r\n\r\nexport default ErrorBoundary;"]},"metadata":{},"sourceType":"module"}